src/compiler/binder.ts

```diff
@@ -52,6 +52,31 @@ module ts {
         }
     }
 
+    const enum ContainerFlags {
+        // The current node is not a container, and no container manipulation should happen before 
+        // recursing into it.
+        None = 0,
+
+        // The current node is a container.  It should be set as the current container (and block-
+        // container) before recursing into it.  The current node does not have locals.  Examples:
+        //
+        //      Classes, ObjectLiterals, TypeLiterals, Interfaces...
+        IsContainer = 0x01,

```

Comment:
can we use bit shifts instead of hex values? personally I find bit shifts easier to reason about current\subsequent values of the flag since next value of the shift is always +1 to the current one